runtime/j9vm/j7vmi.c

```diff
@@ -2338,6 +2339,18 @@ jobject JNICALL JVM_UnloadLibrary(jint arg0)
 {
 #if JAVA_SPEC_VERSION >= 15
 	Trc_SC_UnloadLibrary_Entry(handle);
+
+#if JAVA_SPEC_VERSION >= 17
+	if (J9_ARE_ALL_BITS_SET((UDATA)handle, J9_OFFLOAD_NATIVE_LIBRARY_TAG)) {
+		J9NativeLibrary *nativeLib = (J9NativeLibrary *)(((UDATA)handle) & ~((UDATA)J9_OFFLOAD_NATIVE_LIBRARY_TAG));
```

Comment:
The suggestion was that you combine "all bits set" with `^`, or "any bits set" with `& ~TAG` (preferring the latter).
The outermost parentheses don't help in `~((UDATA)J9_OFFLOAD_NATIVE_LIBRARY_TAG)` (there no way to confuse the order of two prefix operators).

Reply:
The "any bit" test is slightly less expensive than the "all bits" test (which is why I prefer it).
I don't understand why it wouldn't make sense to you, but where we are now should function properly.